### ISSUE TITLE
Placeholder for tracker->collector hinting on whether to resend or not

### ISSUE THREAD

**Issue by alexanderdean** [snowplow/snowplow#2467]
Only a tracker knows if it has a chance to re-send an event on failure.

After #2465, we could consider some kind of hinting so the tracker can tell the collector to really, really try to send the event - i.e. the tracker is handing off the event to the collector, and the collector has to make best efforts to send it.

**Comment by fblundun** [comment]
At the moment, the tracker only knows whether to hold onto the event based on the status code returned by the collector. Is the idea that the collector would receive the event, quickly try to fully process and send it to Kinesis, and only then respond to the original request with a status code depending on whether the event was sent on successfully?

**Comment by jbeemster** [comment]
This was more for Trackers that have no way of holding onto the event, no SQLite or local storage to hold them in.  In the case of the pixel tracker it needs to be accepted by the collector or it will be lost forever.

So under extreme stress the collector would potentially refuse events that can be stored on their host device but accept events that cannot be kept.

**Comment by alexanderdean** [comment]
Hey @fblundun - this ticket isn't about putting the send to Kinesis in the hot path (i.e. in the request/response transaction). Rather, it is about changing the behaviour in scenario 6 below:

| Name          | Datatype | Format |
|---------------|----------|--------|
| isVip         | bool     | -      |
| lifetimeValue | double   | -      |
| spendSegment  | string   | -      |
| whenCreated   | datetime | epoch  |

**Comment by fblundun** [comment]
I see what you mean!

We could go even further and send a certain status code when the collector is under heavy load. Then any client-side caching tracker receiving that status code would stop sending events for a certain period.

**Comment by alexanderdean** [comment]
Yes! That would be cool. It feels like we need to send more information in both directions:

1. A tracker should tell the collector what its capabilities are (particularly important for very primitive trackers like pixel or Arduino)
2. The collector should tell a tracker what is currently happening (particularly useful for sophisticated trackers like JS/iOS/Android)

We just need to remember that any data from collector to tracker will just relate to the single collector instance that the tracker is talking to via the ELB - it's still useful though...

**Comment by alexanderdean** [comment]
> So under extreme stress the collector would potentially refuse events that can be stored on their host device but accept events that cannot be kept.

This is a nice point - it basically implements a priority system, where "goldfish" :fish: trackers take priority over "elephant" :elephant: trackers...
